### DISCUSSION TITLE
mingw-w64-x86_64-python downgrade

### DISCUSSION THREAD

**Discussion by RicardoSaikali** [msys2/MINGW-packages#18881] [Q&A]
Hello,

I am using gstreamer with pygobject on mingw64. My current mini test application is working with the default python version that is used with mingw-w64-x86_64-python (uses 3.11.6). This mini test app uses gtk and is working fine, but my next step in making a test app will involve adding my other dependencies. For that I would like to use mingw-w64-x86_64-python3 version 3.10.

 When I try to install python 3.10 via pacman -U python-3.10.10-2-x86_64.pkg.tar.zst, I am able to downgrade my python3 version, however pip no longer works and I assume all the packages I downloaded with packman don't either 
(packages like  mingw-w64-x86_64-python3-gobject, mingw-w64-x86_64-python3-pip, mingw-w64-x86_64-gstreamer).
Is there a way to use mingw-w64-x86_64-python 3.10 with my packages on mingw64 easily?

**Comment by Biswa96** [comment]
> Is there a way to use mingw-w64-x86_64-python 3.10 with my packages on mingw64 easily?

Nope 😞

**Reply by RicardoSaikali** [reply]
is there a difficult way at least?

**Reply by Biswa96** [reply]
If you want you can rebuild every python packages. Before that, please wait for pygobject maintainer to comment.

**Reply by RicardoSaikali** [reply]
update: I was downloading  python-3.10.10-2-x86_64.pkg.tar.zst which is the wrong file. I tired with mingw-w64-x86_64-python-3.10.12-1-any.pkg.tar.zst which the python i want, then installed pip by using python-m ensurepip and I am running into an issue installing kivy:

Collecting kivy[base]
  Using cached Kivy-2.2.1.tar.gz (24.0 MB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      Looking in indexes: https://pypi.org/simple, https://kivy.org/downloads/simple/
      Collecting setuptools
        Using cached setuptools-68.2.2-py3-none-any.whl (807 kB)
      Collecting wheel
        Using cached wheel-0.41.2-py3-none-any.whl (64 kB)
      Collecting packaging
        Using cached packaging-23.2-py3-none-any.whl (53 kB)
      Collecting cython!=0.27,!=0.27.2,<=0.29.33,>=0.24
        Using cached Cython-0.29.33-py2.py3-none-any.whl (987 kB)
      ERROR: Could not find a version that satisfies the requirement kivy_deps.gstreamer_dev~=0.3.3 (from versions: none)
      ERROR: No matching distribution found for kivy_deps.gstreamer_dev~=0.3.3

**Comment by wszqkzqk** [comment]
MSYS2 inherits the rolling update feature from the upstream archlinux, so it's not easy to do so.
